galsim/utilities.py

```diff
@@ -418,3 +418,113 @@ def _gammafn(x):
+                               them (False)? [default:True]
+    @param suppress_warnings   Suppresses the warnings about the pixel scale of the output, if True.
+                               [default:False]
+    @param catch_offset_errors Checks for the consistency of `offsets` with `N` and raises Errors
+                               if inconsistencies found (True). Recommended, but could slow down
+                               the routine a little. [default:True]
+
+    @returns the interleaved image
+    """
+
+    if isinstance(N,int):
+        n1,n2 = N,N
+    elif hasattr(N,'__iter__'):
+        if len(N)==2:
+            n1,n2 = N
+        else:
+            raise TypeError("'N' has to be a list or a tuple of two integers")
+        if not (isinstance(n1,int) and  isinstance(n2,int)):
+            raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+
+    if len(im_list)<2:
+        raise TypeError("'im_list' needs to have at least two instances of galsim.Image")
+
+    if (n1*n2 != len(im_list)):
+        raise ValueError("'N' is incompatible with the number of images in 'im_list'")
+
+    if len(im_list)!=len(offsets):
+        raise ValueError("'im_list' and 'offsets' must be lists of same length")
+    for offset in offsets:
+        if not isinstance(offset,galsim.PositionD):
+            raise TypeError("'offsets' must be a list of galsim.PositionD instances")
+
+    if isinstance(im_list[0],galsim.Image):
+        y_size, x_size = im_list[0].array.shape
+        scale = im_list[0].scale
+    else:
+        raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+    for im in im_list[1:]:
+        if not isinstance(im,galsim.Image):
+            raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+        if im.array.shape != (y_size,x_size):
+            raise ValueError("All galsim.Image instances in 'im_list' must be of the same size")
+ 
+        if im.scale != scale:
+            raise ValueError(
+                "All galsim.Image instances in 'im_list' must have the same pixel scale")
+
+    img_array = np.zeros((n2*y_size,n1*x_size))
+    # The tricky part - going from (x,y) Image coordinates to array indices
+    # DX[i'] = -(i+0.5)/n+0.5 = -i/n + 0.5*(n-1)/n
+    #    i  = -n DX[i'] + 0.5*(n-1)
+    for k in xrange(len(offsets)):
+        dx, dy = offsets[k].x, offsets[k].y
+
+        i = int(round((n1-1)*0.5-n1*dx))
+        j = int(round((n2-1)*0.5-n2*dy))
+
+        if catch_offset_errors is True:
+            err_i = (n1-1)*0.5-n1*dx - round((n1-1)*0.5-n1*dx)
+            err_j = (n2-1)*0.5-n2*dy - round((n2-1)*0.5-n2*dy)
+            tol = 1.e-6
+            if abs(err_i)>tol or abs(err_j)>tol:
+                raise ValueError("'offsets' must be a list of galsim.PositionD instances with x "
```

Comment:
I think I'd just report "... values spaced by 1/n1 ..." so it's clear that the constraint depends on n1, and isn't a magic number.